web/src/engine/websites/MangaKawaii.ts

```diff
@@ -0,0 +1,83 @@
+import { Tags } from '../Tags';
+import icon from './MangaKawaii.webp';
+import { Chapter, DecoratableMangaScraper, Manga, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchCSS, FetchRequest, FetchWindowScript } from '../FetchProvider';
+
+const cdn = 'https://cdn.mangakawaii.pics';
+
+const pagesScript = `
+    new Promise((resolve, reject) => {
+        try {
+            resolve(pages.map(page => '${cdn}/uploads/manga/'+ oeuvre_slug +'/chapters_'+applocale+'/'+chapter_slug+ '/'+page.page_image));
+        }
+            catch(error) {
+                reject(error);
+            }
+        });
+`;
+
+@Common.MangaCSS(/^https?:\/\/www\.mangakawaii\.io\/manga\/[^/]+$/, 'div.manga-view__header h1[itemprop="name headline"]')
+@Common.PagesSinglePageJS(pagesScript, 500)
+@Common.ImageAjax()
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('mangakawaii', `MangaKawaii`, 'https://www.mangakawaii.io', Tags.Language.French, Tags.Source.Aggregator, Tags.Media.Manga, Tags.Media.Manhua, Tags.Media.Manhwa);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchMangas(provider: MangaPlugin) : Promise<Manga[]> {
+        const token = await this.getToken();
+        const request = new FetchRequest(new URL('changeMangaList?type=text', this.URI).href);
+        request.headers.set('X-Requested-With', 'XMLHttpRequest');
+        request.headers.set('X-CSRF-TOKEN', token);
+        const data = await FetchCSS<HTMLAnchorElement>(request, 'ul.manga-list-text li a.alpha-link');
+        return data.map(element => {
+            const bloat = element.querySelector('span');
+            if (bloat) element.removeChild(bloat);
+            return new Manga(this, provider, element.pathname, element.text.trim());
+        });
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
```

Comment:
Suggestion to make this method more flexible by providing languages as array and iterate through array `[ 'en', 'fr' ]`